File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripService.java
package org.craftedsw.tripservicekata.trip;

import java.util.List;
import org.craftedsw.tripservicekata.exception.UserNotLoggedInException;
import org.craftedsw.tripservicekata.user.User;
import org.craftedsw.tripservicekata.user.UserSession;

public class TripService {

	public List<Trip> getTripsByUser(User user) throws UserNotLoggedInException {
		return loggedUser().getTripsIfIsFriendWith(user);
	}

	protected User loggedUser() {
		return UserSession.getInstance().getLoggedUser();
	}
}

File: kotlin/trip-service-kata/src/main/kotlin/org/craftedsw/tripservicekata/trip/TripService.kt
package org.craftedsw.tripservicekata.trip

import org.craftedsw.tripservicekata.exception.UserNotLoggedInException
import org.craftedsw.tripservicekata.user.User
import org.craftedsw.tripservicekata.user.UserSession
import java.util.*

open class TripService {

    fun getTripsByUser(user: User): List<Trip> {

        val loggedUser: User = getLoggedUser() ?: throw UserNotLoggedInException()

        if (user.isFriendsWith(loggedUser)) {
            return findTripsFor(user)
        }

        return ArrayList<Trip>()
    }

    open fun findTripsFor(user: User) = TripDAO.findTripsByUser(user)
    open fun getLoggedUser() = UserSession.instance.loggedUser

}
